crowbar_framework/app/models/service_object.rb

```diff
@@ -1408,6 +1431,35 @@ def apply_role(role, inst, in_queue)
     # XXX: This should not be done this way.  Something else should request this.
     system("sudo", "-i", Rails.root.join("..", "bin", "single_chef_client.sh").expand_path.to_s) if !ran_admin
 
+    # are there any roles to remove from the runlist?
+    # The @bcname proposal's elements key will contain the removal intentions
+    # proposal.elements =>
+    # {
+    #   "role1_remove" => ["node1"],
+    #   "role2_remove" => ["node2", "node3"]
+    # }
+    roles_to_remove = databag["deployment"][@bc_name]["elements"].select do |r|
+      r =~ /_remove/
+    end.keys
+    # returns ["role1_remove", "role2_remove"] || {}
+    roles_to_remove.each do |role_to_remove|
+      nodes_with_role_to_remove = databag["deployment"][@bc_name]["elements"][role_to_remove]
+      # returns ["node1", "node2"] || nil
+      # are there any leftover nodes where runlist removal
+      # has not been performed?
+      unless nodes_with_role_to_remove.nil?
```

Comment:
I don't think this can be `nil`, as otherwise, the `select` above wouldn't have returned role. Or can it still be `nil`?